src/hooks/useFetchImages.ts

```diff
@@ -1,71 +1,78 @@
 import { useEffect, useState } from 'react'
+import { useQuery } from '@tanstack/react-query'
 import unixToDate from '../utilities/date-lib/unixToDate'
 import dateToUnix from '../utilities/date-lib/dateToUnix'
 import unixNow from '../utilities/date-lib/unixNow'
 import ky from 'ky'
 
-const useFetchImages = (needMoreImages: boolean) => {
-  const [isLoading, setIsLoading] = useState(false)
-  const [images, setImages] = useState<Image[]>([])
+type UseFetchImagesResult = {
+  isLoading: boolean
+  error?: Error
+  data: Image[]
+}
+
+const useFetchImages = (needMoreImages: boolean): UseFetchImagesResult => {
   const [increment, setIncrement] = useState(1)
   const [startDate, setStartDate] = useState(unixToDate(unixNow - 518400)) // start w/ last 7 days of data
   const [endDate, setEndDate] = useState(unixToDate(unixNow))
 
-  // Updating time frames once week of images has been called
-  useEffect(() => {
-    setEndDate(() => {
-      const startDateAsUnix = dateToUnix(startDate)
-      const dayBeforePrevStart = unixToDate(startDateAsUnix - 43200)
-
-      return dayBeforePrevStart
-    })
-
-    setStartDate((prevStart) => {
-      const prevStartAsUnix = dateToUnix(prevStart)
-      const sevenDaysFromStart = unixToDate(prevStartAsUnix - 518400)
-
-      return sevenDaysFromStart
-    })
-  }, [increment, isLoading])
-
-  // Api Call
-  useEffect(() => {
-    if (!needMoreImages) return
-
-    const requestApiData = async () => {
+  const {
+    isLoading,
+    error,
+    data: images,
+  } = useQuery({
+    enabled: needMoreImages,
+    queryKey: ['images'],
+    queryFn: async () => {
```

Comment:
Here, useQuery accepts a generic that will type your query results. When you do this, you'll need to modify the function arguments to match one of useQuery's other signatures. I think this just helps useQuery internally narrow types for you from the generic. 
The new function should be called like this: `useQuery(queryKey, queryFn, options)`
```suggestion
  } = useQuery(
  ['images'],
  async () => {....}
  {
    enabled: needMoreImages,
   }
```

Reply:
I saw this but just wasn't sure which method was better. But I agree this cleans up the readability. Thanks